qa/tasks/cephfs/test_nfs.py

```diff
@@ -145,11 +150,25 @@ def _test_create_cluster(self):
         '''
         Test single nfs cluster deployment.
         '''
-        # Disable any running nfs ganesha daemon
-        self._check_nfs_server_status()
-        self._nfs_cmd('cluster', 'create', self.cluster_id)
-        # Check for expected status and daemon name (nfs.<cluster_id>)
-        self._check_nfs_cluster_status('running', 'NFS Ganesha cluster deployment failed')
+        with contextutil.safe_while(sleep=4, tries=10) as proceed:
+            while proceed():
+                try:
+                    # Disable any running nfs ganesha daemon
+                    self._check_nfs_server_status()
+                    cluster_create = self._nfs_complete_cmd(
+                        f'cluster create {self.cluster_id}')
+                    if cluster_create.stderr and 'cluster already exists' \
+                            in cluster_create.stderr.getvalue():
+                        self._test_delete_cluster()
+                        continue
+                    # Check for expected status and daemon name
+                    # (nfs.<cluster_id>)
+                    self._check_nfs_cluster_status(
+                        'running', 'NFS Ganesha cluster deployment failed')
+                    break
+                except (AssertionError, CommandFailedError) as e:
+                    log.warning(f'{e}, retrying')
+                    continue
```

Comment:
required?

Reply:
yes, if there is commandfailed error in line 157, 158 or assertionerror in line 166 then we assume there is some intermittent issue which might get resolved the next iteration

Reply:
Sure. But the exception is handled, isn't it?

Reply:
you mean to say `safe_while` will suppress all the exceptions?

Reply:
`continue` was added to make it more readable, it has been removed as it didn't had much of the use.